### PR TITLE
Statechart Model extended, to allow multiple annotations 

### PR DESCRIPTION
**Pull request by LendvaiDenes**

### CODE REVIEW

File: plugins/core/hu.bme.mit.gamma.statechart.model/src/hu/bme/mit/gamma/statechart/derivedfeatures/StatechartModelDerivedFeatures.java [hunk 1]
```diff
@@ -1542,10 +1542,12 @@ public static TimeSpecification getTimeoutValue(TimeoutDeclaration timeout) {
 	}
 	
 	public static Component getMonitoredComponent(StatechartDefinition adaptiveContract) {
-		StatechartAnnotation annotation = adaptiveContract.getAnnotation();
-		if (annotation instanceof AdaptiveContractAnnotation) {
-			AdaptiveContractAnnotation adaptiveContractAnnotation = (AdaptiveContractAnnotation) annotation;
-			return adaptiveContractAnnotation.getMonitoredComponent();
+		List<StatechartAnnotation> annotations = adaptiveContract.getAnnotations();
+		for(StatechartAnnotation annotation: annotations) {
```
Comment (grbeni) [inline plugins/core/hu.bme.mit.gamma.statechart.model/src/hu/bme/mit/gamma/statechart/derivedfeatures/StatechartModelDerivedFeatures.java:1546]: Add a space between `for` and `(`.

File: plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/StatechartGenerator.xtend [hunk 2]
```diff
@@ -382,12 +391,14 @@ class StatechartGenerator extends ScenarioModelSwitch<EObject> {
...
 		} else {
 			handleDelays(set)
-
 			setupForwardTransition(set, first, true, isNegated, forwardTransition)
-
 			forwardTransition.priority = BigInteger.valueOf(3)
-			forwardTransition.guard = getGuard(1, allowedGlobalWaitMin, allowedGlobalWaitMax)
-			forwardTransition.effects.add(setIntVariable(1, 0))
+			
+			if(generationMode != StatechartGenerationMode.GENERATE_ONLY_FORWARD){
```
Comment (grbeni) [inline plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/StatechartGenerator.xtend:397]: Add a space between `if` and `(`.

File: plugins/scenario/hu.bme.mit.gamma.scenario.trace.generator/src/hu/bme/mit/gamma/scenario/trace/generator/ScenarioStatechartTraceGenerator.xtend [hunk 3]
```diff
@@ -31,6 +31,8 @@ import java.io.File
 import java.util.List
 
 import static extension hu.bme.mit.gamma.statechart.derivedfeatures.StatechartModelDerivedFeatures.*
+import hu.bme.mit.gamma.expression.model.ExpressionModelFactory
+import java.math.BigInteger
```
Comment (grbeni) [inline plugins/scenario/hu.bme.mit.gamma.scenario.trace.generator/src/hu/bme/mit/gamma/scenario/trace/generator/ScenarioStatechartTraceGenerator.xtend:35]: Organize imports.

File: tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-18-thread-1.cex [hunk 4]
```diff
@@ -0,0 +1,21 @@
+(XstsStateSequence
```
Comment (grbeni) [inline tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-18-thread-1.cex:1]: Delete file.

File: tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-18-thread-2.cex [hunk 5]
```diff
@@ -0,0 +1,21 @@
+(XstsStateSequence
```
Comment (grbeni) [inline tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-18-thread-2.cex:1]: Delete file.

File: tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-19-thread-1.cex [hunk 6]
```diff
@@ -0,0 +1,11 @@
+(XstsStateSequence
```
Comment (grbeni) [inline tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-19-thread-1.cex:1]: Delete file.

File: tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-19-thread-2.cex [hunk 7]
```diff
@@ -0,0 +1,11 @@
+(XstsStateSequence
```
Comment (grbeni) [inline tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-19-thread-2.cex:1]: Delete file.

File: tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-20-thread-1.cex [hunk 8]
```diff
@@ -0,0 +1,31 @@
+(XstsStateSequence
```
Comment (grbeni) [inline tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-20-thread-1.cex:1]: Delete file.

File: tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-20-thread-2.cex [hunk 9]
```diff
@@ -0,0 +1,31 @@
+(XstsStateSequence
```
Comment (grbeni) [inline tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/.AdaptiveContractCrossroad-pool-20-thread-2.cex:1]: Delete file.

**Review by grbeni — CHANGES_REQUESTED** [review]
I left some minor comments, could you attend to them?

File: tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/n/NormalTrace9.get [hunk 10]
```diff
@@ -0,0 +1,31 @@
+import "/hu.bme.mit.gamma.tutorial.contract.finish/model/Crossroad.gcd"
```
[the remark this answers is not in the document]
Comment (LendvaiDenes) [reply tutorial/hu.bme.mit.gamma.tutorial.contract.finish/model/adaptive/n/NormalTrace9.get:1]: These are the folders containing the execution traces generated for scenarios Blinking, Init and Normal.

**Review by grbeni — APPROVED** [review]

**Pull request merged**